apis/python/tests/test_update_dataframes.py

```diff
@@ -11,81 +11,122 @@
-    with tempfile.TemporaryDirectory() as exp_path:
-        original = adata.copy()
-        tiledbsoma.io.from_anndata(exp_path, adata, measurement_name="RNA")
-        verify_obs_var(original, adata)
-        readback = request.param
-        with tiledbsoma.Experiment.open(exp_path) as exp:
-            o1 = exp.obs.schema
-            v1 = exp.ms["RNA"].var.schema
-            if readback:
+def multiple_fixtures_with_readback(request, conftest_adata) -> TestCase:
+    """
+    Ingest an ``AnnData``to a SOMA ``Experiment``, yielding a ``TestCase`` with the old and new AnnData objects.
+
+    * The input AnnData is from a hard-coded fixture in conftest.py, not specifiable here as an argument
+    * The `request` is nominally for tests to give a boolean for (False) whether they want the
+      `new_obs` and `new_var` to be the same as the input conftest_adata, or (True) whether they want
+      the `new_obs` and `new_var` to be gotten from writing the conftest_adata object via
+      tiledbsoma.io.from_anndata to a SOMA experiment and then read back via the SOMA API and
+      to_pandas() calls.
+    * The slots in `TestCase` will be available to tests which use this multi-fixture:
+      e.g. if the TestCase class has a `foo` slot then a unit-test function in this file can be
+      passed a `foo` argument
+    """
+    with tempfile.TemporaryDirectory() as experiment_path:
+        old_anndata = conftest_adata.copy()
+        tiledbsoma.io.from_anndata(
+            experiment_path, conftest_adata, measurement_name="RNA"
+        )
+
+        # Check that the anndata-to-soma ingestion didn't modify the old_anndata object (which is
+        # passed by reference to the ingestor) while it was doing the ingest
+        verify_obs_and_var_same(old_anndata, conftest_adata)
+
+        use_readback = request.param
+
+        with tiledbsoma.Experiment.open(experiment_path) as exp:
+            obs_schema = exp.obs.schema
+            var_schema = exp.ms["RNA"].var.schema
+            if use_readback:
                 new_obs = exp.obs.read().concat().to_pandas()
                 new_var = exp.ms["RNA"].var.read().concat().to_pandas()
             else:
-                new_obs = adata.obs
-                new_var = adata.var
+                new_obs = conftest_adata.obs
+                new_var = conftest_adata.var
 
             yield TestCase(
-                exp_path=exp_path,
-                original=original,
-                new=adata,
+                experiment_path=experiment_path,
+                old_anndata=old_anndata,
+                new_anndata=conftest_adata,
                 new_obs=new_obs,
                 new_var=new_var,
-                o1=o1,
-                v1=v1,
+                obs_schema=obs_schema,
+                var_schema=var_schema,
             )
 
 
-def create_member_fixture(name):
-    """Create a ``pytest.fixture`` for a ``TestCase`` field."""
+for field in fields(TestCase):
```

Comment:
I think this block was a main culprit making things too magical; unrolled it into 7 features (one for each `TestCase` field) in #2529.